pycloudlib/gce/cloud.py

```diff
@@ -98,19 +109,30 @@ def delete_image(self, image_id):
         Args:
             image_id: string, id of the image to delete
         """
-        raise NotImplementedError
+        api_image_id = self.compute.images().get(
+            project=self.project,
+            image=os.path.basename(image_id)
+        ).execute()['id']
+        response = self.compute.images().delete(
+            project=self.project,
+            image=api_image_id,
+        ).execute()
 
-    def get_instance(self, instance_id):
+        raise_on_error(response)
+
+    def get_instance(self, instance_id, project=None, zone=None, name=None):
         """Get an instance by id.
 
         Args:
-            instance_id:
+            instance_id: The instance ID returned upon creation
+            project: Project used when creating this instance
+            zone: Zone used when creating this instance
 
         Returns:
             An instance object to use to manipulate the instance further.
 
         """
-        raise NotImplementedError
+        return GceInstance(self.key_pair, instance_id, project, zone, name)
```

Comment:
We need to update this callsite to source self.project/self.region/self.zone if not provided.

Reply:
Source from where? I figured it'd be invalid if they're not provided.

There's `None` checking for it when the instance is created, so I didn't think we'd need to also have the checking here.
https://github.com/canonical/pycloudlib/pull/27/files#diff-d800499e0c4918d48411248888644f8aR27

Reply:
@TheRealFalcon This question really is, are you intending to support project/zone overrides in GCE.get_instance?
We already persist the project and zone on the GCE.BaseCloud instance, maybe we should make those params mandatory in __init__ and we don't have to surface them here too. Or maybe you intended to make get_instance allow for default overrides of project and zone in case a user wanted to switch zones using an already initialized GCE cloud?.

Reply:
Ah, I see what you're saying now. I wasn't intending they'd be overridden in the `get_instance` call, so it makes sense to use the instance variables rather than passing them there. I think I was thinking of it as some kind of static-like method, even though I don't think we want that. I'll fix it